src/graph/impl/KokkosGraph_Distance2Color_impl.hpp

```diff
@@ -849,13 +849,18 @@ class GraphColorD2
       nnz_lno_t vid      = _vertexList(vid_);
       color_t   my_color = _colors(vid);
 
+      // If my_color is 0 then we should bail here b/c vertex was never colored.
+      if(my_color==0) return;
```

Comment:
This case should never be hit. Even when my_color is 0, I think  the correct thing is to place it in the recolorlist. 
Simply, we need to remove this line.